File: cpp/points_object/tests/test_point.cpp
#include "test_helper.hpp"
#include "../src/point.hpp"

TEST(test_point_stores_coords) {
    Point p{3, 4};
    ASSERT_EQ(p.coords().size(), 2u);
    ASSERT_EQ(p.coords()[0], 3.0);
    ASSERT_EQ(p.coords()[1], 4.0);
}

TEST(test_point_is_any_dimension) {
    ASSERT_EQ((Point{1}).dimension(), 1u);
    ASSERT_EQ((Point{1, 2}).dimension(), 2u);
    ASSERT_EQ((Point{1, 2, 3, 4, 5}).dimension(), 5u);
}

TEST(test_point_equality) {
    ASSERT_TRUE((Point{1, 2}) == (Point{1, 2}));
    ASSERT_TRUE((Point{1, 2}) != (Point{2, 1}));
}

void run_point_tests() {
    std::cout << "test_point:\n";
    RUN(test_point_stores_coords);
    RUN(test_point_is_any_dimension);
    RUN(test_point_equality);
}

File: python/object_python/points_object/tests/test_point.py
from src import Point


def test_point_stores_coords():
    assert Point(3, 4).coords == (3, 4)


def test_point_is_any_dimension():
    assert Point(1).dimension == 1
    assert Point(1, 2).dimension == 2
    assert Point(1, 2, 3, 4, 5).dimension == 5


def test_point_equality():
    assert Point(1, 2) == Point(1, 2)
    assert Point(1, 2) != Point(2, 1)


def test_point_is_hashable():
    assert {Point(1, 2), Point(1, 2)} == {Point(1, 2)}
